File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        det = {0:0}
        s = 0
        for i in range(len(nums)):
            s += nums[i]
            if s%k not in det:
                det[s%k] = i + 1
            elif det[s%k] < i:
                return True
        return False

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> hashMap{{0, 0}};
        
        int sum=0;
        for(int i=0; i<nums.size(); i++){
            sum += nums[i];
            if(!hashMap.count(sum%k)){
                hashMap[sum%k] = i+1;
            }else if(hashMap[sum%k] < i){
                return true;
            }
        }
        return false;
    }
};